### PR TITLE
Call to action

### PR DESCRIPTION
**Pull request by uvluni**
Clicking on the Call To Action (on a card from the grid) updates the homepage state and toggles a modal for booking a seat in the restaurant (TBD)

### CODE REVIEW

File: src/pages/HomePage/HomePage.js [hunk 1]
```diff
@@ -31,7 +44,17 @@ class HomePage extends Component {
...
+            filteredCards={filteredCards}
+            {...data}
+            toggleCTADialog={this.toggleCTADialog}
+          />
+          <CallToActionDialog
+            selectedRest={this.state.selectedRest}
+            open={this.state.toggleCTADialog}
+            toggleCTADialog={this.toggleCTADialog}
+          />
+
```
Comment (Alaev) [inline src/pages/HomePage/HomePage.js]: Remove empty line

File: src/pages/HomePage/HomePage.js [hunk 2]
```diff
@@ -4,20 +4,33 @@ import { Main, Search, Left } from './HomePage.styles';
...
 import BottomLeft from '../../components/FeaturedCard/FeaturedCard';
+import CallToActionDialog from '../../components/CallToActionDialog/CallToActionDialog';
 
 class HomePage extends Component {
   constructor() {
     super();
     const filteredCards = data.cards;
     const { cards } = data;
-    this.state = { cards, filteredCards };
+    this.state = { cards, filteredCards, toggleCTADialog: false, selectedRest: '' };
```
Comment (Alaev) [inline src/pages/HomePage/HomePage.js]: set selectedRest to `null` instead of empty string.

File: src/pages/HomePage/HomePage.js [hunk 3]
```diff
@@ -4,20 +4,33 @@ import { Main, Search, Left } from './HomePage.styles';
...
     this.updateFilterCards = this.updateFilterCards.bind(this);
+    this.toggleCTADialog = this.toggleCTADialog.bind(this);
   }
 
   updateFilterCards(fl) {
     this.setState({ filteredCards: fl });
   }
 
+  toggleCTADialog(restId) {
+    const previosState = this.state.toggleCTADialog;
```
Comment (Alaev) [inline src/pages/HomePage/HomePage.js]: typo
change to `previousState`

File: src/components/CallToActionDialog/CallToActionDialog.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import Dialog from 'material-ui/Dialog';
+import FlatButton from 'material-ui/FlatButton';
+
```
Comment (Alaev) [inline src/components/CallToActionDialog/CallToActionDialog.js]: remove empty line

File: src/components/CallToActionDialog/CallToActionDialog.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
```
Comment (Alaev) [inline src/components/CallToActionDialog/CallToActionDialog.js]: remove empty line

File: src/components/CardWithImage/CardWithImage.js [hunk 6]
```diff
@@ -2,7 +2,7 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import { Card, CardActions, CardMedia, CardText } from 'material-ui/Card';
 import FlatButton from 'material-ui/FlatButton';
-import { red100 } from 'material-ui/styles/colors';
+import { red100, deepPurple100 } from 'material-ui/styles/colors';
```
Comment (Alaev) [inline src/components/CardWithImage/CardWithImage.js]: remove empty line

File: src/pages/HomePage/HomePage.js [hunk 7]
```diff
@@ -11,7 +11,7 @@ class HomePage extends Component {
     super();
     const filteredCards = data.cards;
     const { cards } = data;
-    this.state = { cards, filteredCards, toggleCTADialog: false, selectedRest: '' };
+    this.state = { cards, filteredCards, toggleCTADialog: false, selectedRest: undefined };
```
Comment (Alaev) [inline src/pages/HomePage/HomePage.js]: did you try to put `null` instead of `undefined`?

Comment (uvluni) [reply src/pages/HomePage/HomePage.js]: Yes,

"TypeError: Cannot read property 'title' of null"

On line 21, the first iteration is trying to access .title of null (selectedRest.title) and crashes.
null.somthing is not valid while undefined.something is ok.

Comment (Alaev) [reply src/pages/HomePage/HomePage.js]: I see,
do you know why we get this error?
```
Failed prop type: Invalid prop `id` of type `number` supplied to `CardWithImage`, expected `string`.
```

**Comment by uvluni** [comment]
It's Fundrey's fault :grinning: 
We had a discssution: are we using numeric IDs or String ones?

**Comment by Alaev** [comment]
I think strings as the server will send `uuid` or `object id`

**Pull request merged**